worker/worker.py

```diff
@@ -285,6 +289,48 @@ def worker(worker_info, password, remote):
     return success
 
 
+def gcc_version():
+    """Parse the output of g++ -E -dM -"""
+    p = subprocess.Popen(
+        ["g++", "-E", "-dM", "-"],
+        stdout=subprocess.PIPE,
+        stdin=subprocess.PIPE,
+        universal_newlines=True,
+        bufsize=1,
+        close_fds=not IS_WINDOWS,
+    )
+
+    p.stdin.close()
+    for line in iter(p.stdout.readline, ""):
+        if "__GNUC__" in line:
+            major = line.split(" ")[2].strip()
```

Comment:
IMO could be simplified in `major = line.split()[2]`
whitespace is the default separator for split() and split() delete all whitespaces
```
>>> a = "   aaa   bbb   ccc   ddd   "
>>> a
'   aaa   bbb   ccc   ddd   '
>>> a.split()
['aaa', 'bbb', 'ccc', 'ddd']
```
split and strip are used when splitting for something different from the space
https://stackoverflow.com/questions/4071396/split-by-comma-and-strip-whitespace-in-python

Reply:
The previous split syntax is dangerous :)
```
>>> a = "   aaa   bbb   ccc   ddd   "
>>> a.split(" ")
['', '', '', 'aaa', '', '', 'bbb', '', '', 'ccc', '', '', 'ddd', '', '', '']
```